### PR TITLE
feat: add customization options for lat/lon or s2 column in maps

### LINKED ISSUE #1029: Maps: Ability to select tag for s2_cell_id
S2_cell_id stored as tag value ~or field values~
given i have data with the s2_cell_id calculated and stored as a tag value
when i attempt to display that data on a map
then i would like the ability to choose that s2_cell_id to use as the lat and lon values

~given i have data with the s2_cell_id calculated and stored as a field value~
~when i attempt to display that data on a map~
~then i would like the ability to choose that s2_cell_id field to use as the lat and lon values~

### PR DESCRIPTION
**Pull request by Palakp41**
Closes #1029 #1027 

This pr adds the ability to allow users to pick the lat/lon or s2 columns in the customization options.

Kapture 2021-06-22 at 15 44 00

### CODE REVIEW

**Review by TCL735 — APPROVED** [review]

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 51 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=master)

File: src/shared/utils/vis.ts [hunk 1]
```diff
@@ -456,3 +461,115 @@ export const getDetectCoordinatingFields = (table: Table) => {
...
+  )
+
+  if (
+    coordinateColumn === CoordinateType.S2 ||
+    coordinateColumn === CoordinateType.Tags
+  ) {
+    return false
+  }
+  return true
+}
```
Comment (hoorayimhelping) [inline src/shared/utils/vis.ts:575]: question, what is the `flagged` part of the methods? I was trying to see where they reference feature flags but couldn't find it

**Review by hoorayimhelping — COMMENTED** [review]
I don't have any blocking stuff, more general comments. We can follow up on them or do them now. I do think that writing some tests for this will make future maintenance easier and you should consider that.

**Comment by Palakp41** [comment]
@hoorayimhelping thank you for your review! I will add these suggestions in the next pr :)

**Pull request merged**